### ISSUE TITLE
appbar with QrScannerOverlayShape Incompatible

### ISSUE THREAD

**Issue by mx4g** [juliuscanute/qr_code_scanner#52]
![image](https://user-images.githubusercontent.com/7499986/67268814-04b3e800-f4e8-11e9-9e06-d475ff695691.png)
The QrScannerOverlayShape  is running up

**Comment by mx4g** [comment]
how to set QrScannerOverlayShape  center in QRView

**Comment by Nirvana-cn** [comment]
The overlay is in the center of QRView by default. But I want move the position of overlay. What are you doing?

![s1](https://user-images.githubusercontent.com/26457604/67286207-908b3b80-f50b-11e9-9d3b-652600eb8d05.png)

```
import 'package:flutter/material.dart';
import 'package:qr_code_scanner/qr_code_scanner.dart';
import 'package:qr_code_scanner/qr_scanner_overlay_shape.dart';

class QRViewExample extends StatefulWidget {
  const QRViewExample({
    Key key,
  }) : super(key: key);

  @override
  State<StatefulWidget> createState() => _QRViewExampleState();
}

class _QRViewExampleState extends State<QRViewExample> {
  var qrText = "";
  QRViewController controller;
  final GlobalKey qrKey = GlobalKey(debugLabel: 'QR');

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("QR Code"),
      ),
      body: Column(
        children: <Widget>[
          Expanded(
            child: QRView(
              key: qrKey,
              onQRViewCreated: _onQRViewCreated,
              overlay: QrScannerOverlayShape(
                borderColor: Colors.red,
                borderRadius: 0,
                borderLength: 30,
                borderWidth: 10,
                cutOutSize: 300,
              ),
            ),
            flex: 1,
          ),
        ],
      ),
    );
  }

  void _onQRViewCreated(QRViewController controller) {
    this.controller = controller;
    controller.scannedDataStream.listen((scanData) {
      controller.pauseCamera();
      setState(() {
        qrText = scanData;
      });
    });
  }

  @override
  void dispose() {
    controller.dispose();
    super.dispose();
  }
}

```

**Comment by mx4g** [comment]
> The overlay is in the center of QRView by default. But I want move the position of overlay. What are you doing?
> 
> ![s1](https://user-images.githubusercontent.com/26457604/67286207-908b3b80-f50b-11e9-9d3b-652600eb8d05.png)
> 
> ```
> import 'package:flutter/material.dart';
> import 'package:qr_code_scanner/qr_code_scanner.dart';
> import 'package:qr_code_scanner/qr_scanner_overlay_shape.dart';
> 
> class QRViewExample extends StatefulWidget {
>   const QRViewExample({
>     Key key,
>   }) : super(key: key);
> 
>   @override
>   State<StatefulWidget> createState() => _QRViewExampleState();
> }
> 
> class _QRViewExampleState extends State<QRViewExample> {
>   var qrText = "";
>   QRViewController controller;
>   final GlobalKey qrKey = GlobalKey(debugLabel: 'QR');
> 
>   @override
>   Widget build(BuildContext context) {
>     return Scaffold(
>       appBar: AppBar(
>         title: Text("QR Code"),
>       ),
>       body: Column(
>         children: <Widget>[
>           Expanded(
>             child: QRView(
>               key: qrKey,
>               onQRViewCreated: _onQRViewCreated,
>               overlay: QrScannerOverlayShape(
>                 borderColor: Colors.red,
>                 borderRadius: 0,
>                 borderLength: 30,
>                 borderWidth: 10,
>                 cutOutSize: 300,
>               ),
>             ),
>             flex: 1,
>           ),
>         ],
>       ),
>     );
>   }
> 
>   void _onQRViewCreated(QRViewController controller) {
>     this.controller = controller;
>     controller.scannedDataStream.listen((scanData) {
>       controller.pauseCamera();
>       setState(() {
>         qrText = scanData;
>       });
>     });
>   }
> 
>   @override
>   void dispose() {
>     controller.dispose();
>     super.dispose();
>   }
> }
> ```

You can add one more Explanded，Because I want to add some button settings.
![20191023175227881](https://user-images.githubusercontent.com/7499986/67381116-f801c380-f5bd-11e9-89f1-a473287b564d.jpg)

**Comment by Mecil9** [comment]
a same question!!!

兄弟你解决了没?

**Comment by mx4g** [comment]
> a same question!!!
> 
> 兄弟你解决了没?

没有呀！只能用Stack来布局了，你也先这样做吧！

**Comment by adihazli** [comment]
Have u guys manage to find out how to add 2 expanded child and adjust overlay shape to center?

**Comment by BelfDev** [comment]
> The overlay is in the center of QRView by default. But I want move the position of overlay. What are you doing?
> 
> ![s1](https://user-images.githubusercontent.com/26457604/67286207-908b3b80-f50b-11e9-9d3b-652600eb8d05.png)
> 
> ```
> import 'package:flutter/material.dart';
> import 'package:qr_code_scanner/qr_code_scanner.dart';
> import 'package:qr_code_scanner/qr_scanner_overlay_shape.dart';
> 
> class QRViewExample extends StatefulWidget {
>   const QRViewExample({
>     Key key,
>   }) : super(key: key);
> 
>   @override
>   State<StatefulWidget> createState() => _QRViewExampleState();
> }
> 
> class _QRViewExampleState extends State<QRViewExample> {
>   var qrText = "";
>   QRViewController controller;
>   final GlobalKey qrKey = GlobalKey(debugLabel: 'QR');
> 
>   @override
>   Widget build(BuildContext context) {
>     return Scaffold(
>       appBar: AppBar(
>         title: Text("QR Code"),
>       ),
>       body: Column(
>         children: <Widget>[
>           Expanded(
>             child: QRView(
>               key: qrKey,
>               onQRViewCreated: _onQRViewCreated,
>               overlay: QrScannerOverlayShape(
>                 borderColor: Colors.red,
>                 borderRadius: 0,
>                 borderLength: 30,
>                 borderWidth: 10,
>                 cutOutSize: 300,
>               ),
>             ),
>             flex: 1,
>           ),
>         ],
>       ),
>     );
>   }
> 
>   void _onQRViewCreated(QRViewController controller) {
>     this.controller = controller;
>     controller.scannedDataStream.listen((scanData) {
>       controller.pauseCamera();
>       setState(() {
>         qrText = scanData;
>       });
>     });
>   }
> 
>   @override
>   void dispose() {
>     controller.dispose();
>     super.dispose();
>   }
> }
> ```

Hi, I made a fork and added a simple property that allows you to customise the cutout bottom offset. In case you wish to check it out, just add the code below to your pubsepc.yaml.

```
qr_code_scanner:
    git:
      url: https://github.com/BelfDev/qr_code_scanner.git
      ref: master
```

**Comment by juliansteenbakker** [comment]
I have merged #115 so in the next release it will be possible to customise the bottom offset.
